testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/javascript/JavascriptAdapterTest.java

```diff
@@ -400,6 +397,69 @@ public void loginRequiredAction() {
 
+    /**
+     * Test for scope handling via explicit Keycloak configuration: <pre>{@code
+     * new Keycloak({.... scope: "profile email phone"})
+     * }</pre>
+     * See KEYCLOAK-14412
+     */
+    @Test
+    public void testOnLoadRequireLoginShouldHonorScopes() {
+
+        JSObjectBuilder config = JSObjectBuilder.create()
+                .add("url", authServerContextRootPage + "/auth")
+                .add("realm", REALM_NAME)
+                .add("clientId", CLIENT_ID)
+                .add("scope", "profile email phone") // phone is optional client scope
+                ;
+
+        JSObjectBuilder initOptions = defaultArguments().loginRequiredOnLoad();
+
+        try {
+            testExecutor.configure(config)
+                        .init(initOptions);
+            // This throws exception because when JavascriptExecutor waits for AsyncScript to finish
+            // it is redirected to login page and executor gets no response
+
+            throw new RuntimeException("Probably the login-required OnLoad mode doesn't work, because testExecutor should fail with error that page was redirected.");
+        } catch (WebDriverException ex) {
+            // should happen
+        }
+
+        testExecutor.loginForm(testUser, this::assertOnTestAppUrl)
+                .init(initOptions, this::assertSuccessfullyLoggedIn)
+        .executeAsyncScript("var callback = arguments[arguments.length - 1];" +
+                "callback(window.keycloak.tokenParsed.scope)", assertOutputContains("phone"));
+    }
+
+    /**
+     * Test for scope handling via Keycloak configuration from {@code keycloak.json} : <pre>{@code
+     * new Keycloak() -> keycloak.json {.... "scope": "profile email phone"}
+     * }</pre>
+     * See KEYCLOAK-14412
+     */
+    @Test // TODO findout why this test fails
+    public void testConfigFromKeycloakJsonShouldHonorScopes() {
+
+        JSObjectBuilder initOptions = defaultArguments().loginRequiredOnLoad();
+
+        try {
+            testExecutor.configure() // use keycloak.json from testsuite/integration-arquillian/servers/auth-server/services/testsuite-providers/src/main/resources/javascript/keycloak.json
+                    .init(initOptions);
+            // This throws exception because when JavascriptExecutor waits for AsyncScript to finish
+            // it is redirected to login page and executor gets no response
+
+            throw new RuntimeException("Probably the login-required OnLoad mode doesn't work, because testExecutor should fail with error that page was redirected.");
+        } catch (WebDriverException ex) {
+            // should happen
+        }
+
+        testExecutor.loginForm(testUser, this::assertOnTestAppUrl)
+                .init(initOptions, this::assertSuccessfullyLoggedIn)
+                .executeAsyncScript("var callback = arguments[arguments.length - 1];" +
+                        "callback(window.keycloak.tokenParsed.scope)", assertOutputContains("phone"));
+    }
+
```

Comment:
Could you add a test also for the `init` function?

Reply:
There are now two tests, one for `initOptions` and another one `loginOptions`.